data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfiguration.java

```diff
@@ -93,10 +86,13 @@ public SearchConfiguration getSearchConfiguration() {
         return searchConfiguration;
     }
 
-    @AssertTrue(message = "Either username and password, or aws options must be specified. Both cannot be set at once.")
-    boolean validateAwsConfigWithUsernameAndPassword() {
-        return !((Objects.nonNull(awsAuthenticationOptions) && (Objects.nonNull(username) || Objects.nonNull(password))) ||
-                (Objects.isNull(awsAuthenticationOptions) && (Objects.isNull(username) || Objects.isNull(password))));
+    void validateAwsConfigWithUsernameAndPassword() {
+
+        if (((Objects.nonNull(awsAuthenticationOptions) && ((Objects.nonNull(username) || Objects.nonNull(password)) || disableAuthentication)) ||
+                (Objects.nonNull(username) || Objects.nonNull(password)) && disableAuthentication) ||
+                (Objects.isNull(awsAuthenticationOptions) && (Objects.isNull(username) || Objects.isNull(password)) && !disableAuthentication)) {
```

Comment:
Minor: it's easier to get a sense of the conditionals by breaking this into a few booleans. Something like
```
final boolean isAwsAuthValid = ..
final boolean is userNamePasswordValid = ..

if (isAwsAuthValid && !userNamePasswordValid ...)
```